### DISCUSSION TITLE
How to load mappings with plugins in v2.5?

### DISCUSSION THREAD

**Discussion by SirenGlitch** [NvChad/NvChad#2983] [Q&A] [answered]
Sorry if this is a really noob question but I can't seem to find any full reference in any issues/discussions.

In v2.0, you could make it so that certain mappings would only load when certain plugins were loaded, using code like this: 
```lua
  {
    "mfussenegger/nvim-dap",
    config = function(_, opts)
      require("core.utils").load_mappings("dap")
    end
  },
```
but in v2.5 I can't seem to find a way to do it, seeing as the `load_mappings()` function is deprecated

**Comment by siduck** [comment, marked as answer]
just put them in your mappings file , no need to use core.utils

**Reply by SirenGlitch** [reply]
Would that not make them load on startup?

This is the mappings.lua file I'm trying to use but I attempted to migrate it to the v2.5 syntax (see below) but couldn't figure out how `M.dap` would be migrated
```lua
local M = {}

M.dap = {
  plugin = true,
  n = {
    ["<leader>db"] = {"<cmd> DapToggleBreakpoint <CR>"}
  }
}
return M
```
Migrated to
```lua
require "nvchad.mappings"

local map = vim.keymap.set

-- DAP mappings
map("n", "<leader>db", "<cmd>DapToggleBreakpoint<CR>", { desc = "Debug Add breakpoint to current line" })
```

**Reply by siduck** [reply]
@SirenGlitch code in a plugin's init function also runs on startup!

**Reply by SirenGlitch** [reply]
> code in a plugin's init function also runs on startup!

So the mappings would be mapped on startup even on v2.0 using `core.utils`?

**Reply by siduck** [reply]
yes

**Reply by SirenGlitch** [reply]
Ahh, okay. Thanks for the help!

**Reply by SirenGlitch** [reply]
What about the `plugin = true` argument from v2.0? From what I see that only loads mappings when explicitly told to load them (i.e. using `load_mappings()`). Was this whole behaviour deprecated with v2.5 or am I misunderstanding?

**Reply by siduck** [reply]
yes it'd load when needed

you dont need it anymore, just store ur mappings somewhere and load it by require

**Reply by batyrmuhadov** [reply]
btw is there any way to look mappings up with telescope like in ohter distros?

**Reply by SirenGlitch** [reply]
@batyrmuhadov if other distros have it, then telescope is capable of it. give [nvim-telescope/telescope.nvim](https://github.com/nvim-telescope/telescope.nvim) a look because it should have some docs

**Reply by siduck** [reply]
obviously! Telescope keymaps command

or press <leader> ch or NvCheatsheet command
